spring-cloud-contract-tools/spring-cloud-contract-spec-pact/src/main/groovy/org/springframework/cloud/contract/verifier/spec/pact/BodyConverter.groovy

```diff
@@ -0,0 +1,130 @@
+package org.springframework.cloud.contract.verifier.spec.pact
+
+import au.com.dius.pact.consumer.dsl.DslPart
+import au.com.dius.pact.consumer.dsl.PactDslJsonArray
+import au.com.dius.pact.consumer.dsl.PactDslJsonBody
+import groovy.json.JsonException
+import groovy.json.JsonSlurper
+import groovy.transform.CompileStatic
+import groovy.transform.PackageScope
+import org.springframework.cloud.contract.spec.internal.Body
+import org.springframework.cloud.contract.spec.internal.DslProperty
+import org.springframework.cloud.contract.verifier.util.ContentUtils
+
+/**
+ * @author Tim Ysewyn
+ * @Since 2.0.0
+ */
+@CompileStatic
+@PackageScope
+class BodyConverter {
+
+	private static final JsonSlurper jsonSlurper = new JsonSlurper()
+
+	static DslPart toPactBody(Body body, Closure dslPropertyValueExtractor) {
+		return traverse(body, null, dslPropertyValueExtractor)
+	}
+
+	static DslPart toPactBody(DslProperty dslProperty, Closure dslPropertyValueExtractor) {
+		return traverse(dslProperty, null, dslPropertyValueExtractor)
+	}
+
+	private static DslPart traverse(Object value, DslPart parent, Closure dslPropertyValueExtractor) {
+		boolean isRoot = parent == null
+		Object v = value
+
```

Comment:
It's just a matter of preference, but could you get rid of all of those separations between blocks? If not, then I'll do it later

Reply:
Consider it done